state/block_test.go

```diff
@@ -0,0 +1,75 @@
+package state
+
+import (
+	"fmt"
+	"math/big"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/common"
+	"github.com/ChainSafe/gossamer/core/types"
+)
+
+func TestGetBlockByNumber(t *testing.T) {
+	dataDir := "../test_data/block"
+
+	//Create a new blockState
+	blockState, err := NewBlockState(dataDir)
+	if err != nil {
+		t.Fatal(err)
+	}
+	//Close DB & erase data dir contents
+	defer func() {
+		err = blockState.Db.Db.Close()
```

Comment:
Should we create a `Stop` method on `blockState` that includes `Db.Db.Close` and `os.RemoveAll` to clean this up and also to make it reusable in multiple tests?

Reply:
this same test is in the #503 , I've added comments there with snippets for the fix, could this be applied there instead ?

Reply:
this has not been addressed but let's add it in #503 when we add multiple tests